source/common/quic/envoy_quic_client_stream.h

```diff
@@ -78,6 +78,10 @@ class EnvoyQuicClientStream : public quic::QuicSpdyClientStream,
   Http::ResponseDecoder* response_decoder_{nullptr};
 
   bool decoded_1xx_{false};
+
+  // When an Http Upgrade is requested, this contains the protocol upgrade type, e.g. "websocket".
```

Comment:
Is the value of `upgrade_type_` the name of the protocol? I think that's what goes in the HTTP Upgrade header. If so, perhaps `upgrade_protocol_` would be a slightly clearer name and perhaps the comment should be:

```
// When an HTTP Upgrade is requested, this fields contains the name of protocol, e.g. "websocket".
```